File: CyclicRotation.js
function solution(arr, k) {
	let arr_length = arr.length;

	if (arr.length == 0) return arr;

	while (arr_length < k) {
		k = k - arr_length;
	}

	let temp_arr = [];
	for (i = 0; i < k; i++) {
		let element = arr.pop();
		temp_arr.push(element);
	}

	temp_arr.reverse();
	return temp_arr.concat(arr);
}

File: CyclicRotation.py
def solution(arr, k):

    if (arr == [] or k == 0): return arr

    while(k > len(arr)):
        k = k - len(arr)
        pass

    if (k == len(arr)): return arr

    return arr[len(arr)-k:] + arr[0:len(arr)-k]


    pass
